lib/Conversion/HWArithToHW/HWArithToHW.cpp

```diff
@@ -102,6 +102,63 @@ static Value extendTypeWidth(OpBuilder &builder, Location loc, Value value,
   return extOp->getOpResult(0);
 }
 
+static bool isSignednessType(Type type) {
+  auto match =
+      llvm::TypeSwitch<Type, bool>(type)
+          .Case<IntegerType>([](auto type) { return !type.isSignless(); })
```

Comment:
Note for future improvement: we'll want this for arbitrary types in the future. E.g. one might want to lower the contents of a channel before lowering away the channel. I'd be nice to find a way to define arbitrary queries on arbitrary types to that we can support other things (e.g. bit width). The trick for this usage is how do we reconstruct a type we don't know anything about...